File: src/main/java/de/tum/www1/orion/util/OrionStudentExerciseRegistry.java
package de.tum.www1.orion.util;

import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

public interface OrionStudentExerciseRegistry extends OrionExerciseRegistry {
    static OrionStudentExerciseRegistry getInstance(@NotNull Project project) {
        return ServiceManager.getService(project, OrionStudentExerciseRegistry.class);
    }
}

File: src/main/java/de/tum/www1/orion/util/registry/OrionStudentExerciseRegistry.kt
package de.tum.www1.orion.util.registry

import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.project.Project

interface OrionStudentExerciseRegistry : OrionExerciseRegistry {
    companion object {
        @JvmStatic
        fun getInstance(project: Project): OrionStudentExerciseRegistry {
            return ServiceManager.getService(project, OrionStudentExerciseRegistry::class.java)
        }
    }
}
